aqua/LyX/Portfile

```diff
@@ -6,9 +6,60 @@ PortGroup           boost 1.0
 
 name                LyX
 conflicts           LyX1
-version             2.4.2
-revision            0
-set branch          [join [lrange [split ${version} .] 0 1] .]
+
+# 2.4.x has expermintal support for Qt6 - we are ignoring ATM
+if {${os.platform} eq "darwin" && ${os.major} < 11} {
+    PortGroup qt4 1.0
+
+    version         2.3.8
+    revision        0
+
+    checksums       rmd160  80b2c53d12b04dd64a6326b4b1ea505dcb352120 \
+                    sha256  7e8f56d148cb459eb00abbe9c6371744e4ff077b842c0024153a3f094b10d4ad \
+                    size    16412700
+
+    set branch      [join [lrange [split ${version} .] 0 1] .]
+
+    configure.pre_args  --prefix=${applications_dir}/LyX.app
+    configure.args-append \
+                    --without-x \
+                    --without-included-boost \
+                    --enable-optimization=-Os \
+                    --with-version-suffix=-${branch} \
```

Comment:
There's some code duplication. These lines seem to be the same regardless of version. Can this common code be pulled out of the conditional?

An additional question is why `-Os` is being hardcoded. Why not use `${configure.optflags}`?